stress/n_workers_reputers__stress_test.go

```diff
@@ -294,86 +414,114 @@ func WorkerReputerLoop(m TestMetadata, topicId uint64) {
-		if err != nil {
-			fmt.Println("Error funding reputer address: ", reputerAddress, " - ", err)
-			continue
+		createNewWorkerAccount := func() {
+			// Generate new worker accounts
+			workerAccountName := "stressWorker" + strconv.Itoa(len(workerAddresses)) + "_topic" + strconv.Itoa(int(topicId))
+			report("Creating worker address: ", workerAccountName)
+			workerAccount, _, err := m.n.Client.AccountRegistry.Create(workerAccountName)
+			if err != nil {
+				report("Error creating worker address: ", workerAccountName, " - ", err)
+				return
+			}
+			workerAddress, err := workerAccount.Address(params.HumanCoinUnit)
+			if err != nil {
+				report("Error getting worker address: ", workerAccountName, " - ", err)
+				return
+			}
+			err = fundAccount(m, topicFunderAccount, topicFunderAddress, workerAddress, 100000)
+			if err != nil {
+				report("Error funding worker address: ", workerAddress, " - ", err)
+				return
+			}
+			err = RegisterWorkerForTopic(m, workerAddress, workerAccount, topicId)
+			if err != nil {
+				report("Error registering worker address: ", workerAddress, " - ", err)
+				return
+			}
+			workerAddresses[workerAccountName] = workerAddress
 		}
-		err = RegisterReputerForTopic(m, reputerAddress, reputerAccount, topicId)
-		if err != nil {
-			fmt.Println("Error registering reputer address: ", reputerAddress, " - ", err)
-			continue
+
+		createNewReputerAccount := func() {
+			// Generate new reputer account
+			reputerAccountName := "stressReputer" + strconv.Itoa(len(reputerAddresses)) + "_topic" + strconv.Itoa(int(topicId))
+			report("Creating reputer address: ", reputerAccountName)
+			reputerAccount, _, err := m.n.Client.AccountRegistry.Create(reputerAccountName)
+			if err != nil {
+				report("Error creating reputer address: ", reputerAccountName, " - ", err)
+				return
+			}
+			reputerAddress, err := reputerAccount.Address(params.HumanCoinUnit)
+			if err != nil {
+				report("Error getting reputer address: ", reputerAccountName, " - ", err)
+				return
+			}
+			err = fundAccount(m, topicFunderAccount, topicFunderAddress, reputerAddress, 100000)
+			if err != nil {
+				report("Error funding reputer address: ", reputerAddress, " - ", err)
+				return
+			}
+			err = RegisterReputerForTopic(m, reputerAddress, reputerAccount, topicId)
+			if err != nil {
+				report("Error registering reputer address: ", reputerAddress, " - ", err)
+				return
+			}
+			err = StakeReputer(m, topicId, reputerAddress, reputerAccount, stakeToAdd)
+			if err != nil {
+				report("Error staking reputer address: ", reputerAddress, " - ", err)
+				return
+			}
+			reputerAddresses[reputerAccountName] = reputerAddress
 		}
-		err = StakeReputer(m, topicId, reputerAddress, reputerAccount, stakeToAdd)
-		if err != nil {
-			fmt.Println("Error staking reputer address: ", reputerAddress, " - ", err)
-			continue
+
+		if i == 0 {
+			createNewWorkerAccount()
+			createNewReputerAccount()
+		} else {
```

Comment:
is this needed? don't we want to add `workersPerEpoch`-many workers, etc. in each epoch including the first one?

Reply:
This is because we want to start with one worker and one reputer even if `workersPerEpoch` and `reputersPerEpoch` are set to 0